pkg/init/init.go

```diff
@@ -234,7 +234,15 @@ func (o *InitClient) PersonalizeName(devfile parser.DevfileObj, flags map[string
 	} else {
 		backend = o.flagsBackend
 	}
-	return backend.PersonalizeName(devfile, flags)
+	name, err := backend.PersonalizeName(devfile, flags)
+	if err != nil {
+		return "", err
+	}
+	err = dfutil.ValidateK8sResourceName("name", name)
+	if err != nil {
+		return "", err
+	}
+	return name, nil
```

Comment:
Could we have an infinite loop somewhere, to repeat asking the name until the syntax is ok, instead of to quit when an error occurs?